### ISSUE TITLE
Building documentation issues

### ISSUE THREAD

**Issue by remi-kazeroni** [ESMValGroup/ESMValCore#1974]
Would anyone know why:
- it takes about 10 minutes to build the documentation on ReadTheDocs after each commit? I remember that it used to be much faster a while ago
- I can't build the documentation locally in a fresh new environment (with Sphinx v6.1.3) via `sphinx-build doc doc/build` or `sphinx-build -Ea doc doc/build` as recommended in our [docs](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#documentation)? I get:

```bash
Notebook error:
PandocMissing in build/.doctrees/nbsphinx/build/.doctrees/nbsphinx/notebooks/composing-recipes.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
https://pandoc.org/installing.html
```
Doing `pip install pandoc` does not help... It would be good to improve the situation for the next release.

**Comment by remi-kazeroni** [comment]
In comparison, the documentation build for the Tool works fine locally, making it easier to work on the docs. I wonder what is different between the Tool and the Core in that respect.

**Comment by valeriupredoi** [comment]
I'll have a look around, Remi, bear with me for a few - good spotting this though!

**Comment by valeriupredoi** [comment]
note that that error occurs on the RTD build page too, only that the build skips it there

**Comment by valeriupredoi** [comment]
I managed to build docs with the latest sphinx=6.1.3 but I am getting a few warnings from the Notebook:
```
/home/valeriu/ESMValCore/doc/notebooks/composing-recipes.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/composing-recipes.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/composing-recipes.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/composing-recipes.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/composing-recipes.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/discovering-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
/home/valeriu/ESMValCore/doc/notebooks/loading-and-processing-data.ipynb:: WARNING: Pygments lexer name 'ipython3' is not known
```

@bouweandela you know a lot more about ipython stuff - what this? @remi-kazeroni see what sphinx deps you have in your env, I got these that do the trick:
```
(release280rc1) valeriu@valeriu-PORTEGE-Z30-C:~/ESMValCore$ conda list sphinx
# packages in environment at /home/valeriu/miniconda3/envs/release280rc1:
#
# Name                    Version                   Build  Channel
nbsphinx                  0.9.1              pyhd8ed1ab_0    conda-forge
sphinx                    6.1.3              pyhd8ed1ab_0    conda-forge
sphinx_rtd_theme          1.2.0              pyha770c72_0    conda-forge
sphinxcontrib-applehelp   1.0.4              pyhd8ed1ab_0    conda-forge
sphinxcontrib-devhelp     1.0.2                      py_0    conda-forge
sphinxcontrib-htmlhelp    2.0.1              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jquery      2.0.0              pyhd8ed1ab_0    conda-forge
sphinxcontrib-jsmath      1.0.1                      py_0    conda-forge
sphinxcontrib-qthelp      1.0.3                      py_0    conda-forge
sphinxcontrib-serializinghtml 1.1.5              pyhd8ed1ab_2    conda-forge
```

**Comment by valeriupredoi** [comment]
pandoc-related warning on RTD raised with RTD here https://github.com/readthedocs/readthedocs.org/issues/10162

**Comment by valeriupredoi** [comment]
pandoc related issue is totally us - and I made a :clown_face: of myself for opening it at RTD - our sphinx is too old (5.0) on the RTD build! See https://github.com/readthedocs/readthedocs.org/issues/10162#issuecomment-1473847526

**Comment by remi-kazeroni** [comment]
Thanks a lot for taking care of this @valeriupredoi! Is that something we should includ in rc2? Or in final v2.8? Just asking from a release point of view. In practice, the sooner this is fixed, the easier it will be to contribute to the docs...

**Comment by bouweandela** [comment]
> it takes about 10 minutes to build the documentation on ReadTheDocs after each commit? I remember that it used to be much faster a while ago

We switched from mocking out most dependencies + installing from PyPI to creating a conda environment. This is much slower, but the big advantage is that we now have one place less where we need to maintain a list of dependencies.

**Comment by bouweandela** [comment]
> I can't build the documentation locally in a fresh new environment (with Sphinx v6.1.3) via sphinx-build doc doc/build or sphinx-build -Ea doc doc/build as recommended in our [docs](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#documentation)?

`mamba install pandoc` fixed it for me, maybe we just need to add `pandoc` to environment.yml in order to build the new Jupyter notebooks as documentation?

**Comment by remi-kazeroni** [comment]
After taking another look at this, I realized that there is no issue building the documentation locally when using an environment created with `ESMValCore/esmvalcore/environment.yml`. The fix #1976 is not really necessary because `Sphinx v6.1.3` is already in the environment.

The issue is when building the Core documentation with the environment from the Tool (`ESMValTool/esmvaltool/environment.yml`). In that case, I get: 
```bash
reading sources... [ 70%] notebooks/composing-recipes
Notebook error:
PandocMissing in notebooks/composing-recipes.ipynb:
Pandoc wasn't found.
Please check that pandoc is installed:
```

> `mamba install pandoc` fixed it for me, maybe we just need to add pandoc to environment.yml in order to build the new Jupyter notebooks as documentation?

For me as well, this fixed my issue. Shall we consider that for v2.8? 
The question is also: is it recommended to build the Core documentation using the Tool env with both Tool and Core installed in dev mode?

**Comment by bouweandela** [comment]
Good find, that explains it!

> is it recommended to build the Core documentation using the Tool env with both Tool and Core installed in dev mode?

Yes, this is how we recommend it in [our documentation](https://docs.esmvaltool.org/en/latest/quickstart/installation.html#using-the-development-version-of-the-esmvalcore-package), so it makes sense to add `pandoc` to the ESMValTool environment with a note that this is needed for building the ESMValCore docs.

**Comment by bouweandela** [comment]
> Shall we consider that for v2.8?

That shouldn't strictly be needed for ESMValTool v2.8 because the ESMValTool environment.yml is used by people developing ESMValTool v2.9-dev..

**Comment by remi-kazeroni** [comment]
Fixed by #1976 and https://github.com/ESMValGroup/ESMValTool/pull/3118
